common/src/Model/Brush.cpp

```diff
@@ -947,8 +970,23 @@ namespace TrenchBroom {
             assert(remaining.vertexCount() + moving.vertexCount() == vertexCount());
             
             // Will vertices be removed?
-            if (!allowVertexRemoval && result.vertexCount() < m_geometry->vertexCount())
-                return false;
+            if (!allowVertexRemoval) {
+                if (result.vertexCount() < m_geometry->vertexCount())
```

Comment:
But at the same time, the remaining checks will not prevent me "picking up" a vertex of another face with a vertex of the face being moved, right? And that's the case I wanted to prevent.